packages/vector_graphics_compiler/lib/src/svg/node.dart

```diff
@@ -0,0 +1,135 @@
+import '../geometry/basic_types.dart';
+import '../geometry/matrix.dart';
+import '../geometry/path.dart';
+import '../paint.dart';
+import '../vector_instructions.dart';
+
+/// A node in a tree of graphics operations.
+///
+/// Nodess describe painting attributes, transformations, paths, and vertices
+/// to draw in depth-first order.
+abstract class Node {
+  /// Constructs a new tree node with
+  const Node({
+    this.id,
+    this.paint,
+    this.parent,
+  });
+
+  final String? id;
+  final Paint? paint;
+  final ParentNode? parent;
+
+  /// Creates a new compatible node with this as if the `newPaint` had
+  /// the current paint applied as a parent.
+  Node adoptPaint(Paint? newPaint);
+
+  void addPaths(VectorInstructions instructions, AffineMatrix transform);
```

Comment:
Can you describe a bit how this interface should be used? What transform do I pass through - should it be different than the transform the node already may have?